.github/workflows/verify_docker_image.yml

```diff
@@ -29,5 +29,11 @@ jobs:
           -w /TheRock \
           ghcr.io/rocm/therock_pytorch_dev_ubuntu_24_04_gfx942:main \
           bash external-builds/pytorch/smoke-tests/run_smoke_tests.sh
+      - name: Waiting for container to finish
+        run: docker wait the_container
+      - name: Inspecting exit code
+        run: exit $(docker inspect the_container --format='{{.State.ExitCode}}')
+      - name: Cleanup container
+        run: docker rm the_container
```

Comment:
I don't think these new steps are actually needed. What error code does the `docker run ...` command in the original step produce when the script fails?

Docs suggest that the exit code of the command you run will be used: https://docs.docker.com/engine/containers/run/#exit-status . See also https://stackoverflow.com/questions/46300610/how-to-get-the-numeric-exit-status-of-an-exited-docker-container.

Reply:
Based on the doc "https://docs.docker.com/engine/containers/run/#exit-status" 125 and 126 are the possible errors we might get. Aren't they?

Reply:
As far as I understand this are only _two_ possibilities. If you can invoke the script and it fails you should get back the error code returned like invoking the bash script without docker at all.

Reply:
Try it and see what happens. You can test with a simple `exit 1` script (or whatever exit code you want) and any docker container, then test again with this specific docker container and smoke test script. When writing any new tests, it helps to see what happens when they succeed and when they fail, so you can trust that they will be useful in the future when test results change.

Reply:
In the smoke test, we have asserts so that makes sure tests are passed. I checked it and it works fine.

I was suspecting that failing the `docker run ` will not be detected without the exit code query however I was wrong. 
One failing case was caught without querying the exit code of the docker run.
```
Run docker run --rm \
unknown flag: --mountt
docker run [OPTIONS] IMAGE [COMMAND] [ARG...]
Run 'docker run --help' for more information
Error: Process completed with exit code 125.
```
CC: @ScottTodd @marbre

Reply:
One more question? should I add `pipefail `to the bash?
`set -xeuo pipefail`